### PR TITLE
Include related zypper logs in fail msg for error 104

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://progress.opensuse.org/issues/70132
Verification: http://kazhua.qa.suse.de/tests/74#step/machinery/38

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -533,7 +533,13 @@ sub zypper_call {
 
     unless (grep { $_ == $ret } @$allow_exit_codes) {
         upload_logs('/var/log/zypper.log');
-        die "'zypper -n $command' failed with code $ret";
+        my $msg = "'zypper -n $command' failed with code $ret";
+        if ($ret == 104) {
+            $msg .= " (ZYPPER_EXIT_INF_CAP_NOT_FOUND)\n\nRelated zypper logs:\n";
+            script_run('tac /var/log/zypper.log | grep -F -m1 -B10000 "Hi, me zypper" | tac | grep \'\(SolverRequester.cc\|THROW\|CAUGHT\)\' > /tmp/z104.txt');
```
Comment (foursixnine) [inline lib/utils.pm:539]: Why not search just for: `SolverRequester.cc.* cap:machinery not found`? however I'm ok with the result, as it captures properly the actual problem and possibly points to the root cause

```suggestion
            script_run('tac /var/log/zypper.log | grep -F -m1 -B10000 "Hi, me zypper" | tac | grep \'SolverRequester.cc .* cap:.* not found\' > /tmp/z104.txt');
```

Comment (asdil12) [reply lib/utils.pm:539]: I would prefer the other way as it provides some additional info.
Also your regex doesn't seem to work: http://kazhua.qa.suse.de/tests/75#step/machinery/38

Comment (foursixnine) [inline lib/utils.pm:539]: One thing I missed, is `tac` available in older systems? as this might affect maintenance

Comment (asdil12) [reply lib/utils.pm:539]: I would guess it is. It is an old part (copyright in sourcecode sais 1988) of coreutils.

Also I found it in this PTF: https://build.suse.de/package/binaries/PTF:9981/coreutils.SUSE_SLE-11_GA/SUSE_SLE-11_GA - that should be old enough I guess.

**Pull request merged**